typings/fengari.d.ts

```diff
@@ -0,0 +1,264 @@
+/*
+ * Typescript type declarations for Fengari
+ */
+
+// tslint:disable:class-name
+
+export const FENGARI_AUTHORS: string;
+export const FENGARI_COPYRIGHT: string;
+export const FENGARI_RELEASE: string;
+export const FENGARI_VERSION: string;
+export const FENGARI_VERSION_NUM: number;
+export const FENGARI_VERSION_MAJOR: string;
+export const FENGARI_VERSION_RELEASE: string;
+export const FENGARI_VERSION_MINOR: string;
+
+export type lua_String = Uint8Array;
+
+export function to_luastring(s: string): lua_String;
+export function to_jsstring(s: lua_String): string;
+export function to_uristring(s: lua_String): string;
+
+export const LUA_OPADD: number;
+export const LUA_OPSUB: number;
+export const LUA_OPMUL: number;
+export const LUA_OPMOD: number;
+export const LUA_OPPOW: number;
+export const LUA_OPDIV: number;
+export const LUA_OPIDIV: number;
+export const LUA_OPBAND: number;
+export const LUA_OPBOR: number;
+export const LUA_OPBXOR: number;
+export const LUA_OPSHL: number;
+export const LUA_OPSHR: number;
+export const LUA_OPUNM: number;
+export const LUA_OPBNOT: number;
+
+export const LUA_TNONE: number;
+export const LUA_TNIL: number;
+export const LUA_TBOOLEAN: number;
+export const LUA_TLIGHTUSERDATA: number;
+export const LUA_TNUMBER: number;
+export const LUA_TSTRING: number;
+export const LUA_TTABLE: number;
+export const LUA_TFUNCTION: number;
+export const LUA_TUSERDATA: number;
+export const LUA_TTHREAD: number;
+
+export namespace lua {
+  interface lua_State {
+  }
+
+  export type lua_Type = number;
+  export type lua_ArithOp = number;
```

Comment:
If you type the `OP` variables down to literals you can then make `lua_ArithOp` a union:

```typescript
export type lua_ArithOp =LUA_OPADD | LUA_OPSUB ... | LUA_OPBNOT;
```